source/MaterialXContrib/MaterialXMaya/AEtemplate/AEMaterialXSurfaceTemplate.mel

```diff
@@ -1,27 +1,65 @@
-global proc MaterialXSurface_extraCtrlsBuild(string $nodeName)
+global proc MaterialXSurface_CommandButtonsNew(string $nodeName)
 {
     string $editCommand = "python(\"from MaterialXNodeUtil import editMaterialXNode; editMaterialXNode(\\\"" + $nodeName + "\\\")\")";
     string $reloadCommand = "reloadMaterialXNode \"" + $nodeName + "\"";
     button -label "Edit" -command $editCommand editButton;
     button -label "Reload" -command $reloadCommand reloadButton;
 }
 
-global proc MaterialXSurface_extraCtrlsUpdate(string $nodeName)
+global proc MaterialXSurface_CommandButtonsReplace(string $nodeName)
 {
     string $editCommand = "python(\"from MaterialXNodeUtil import editMaterialXNode; editMaterialXNode(\\\"" + $nodeName + "\\\")\")";
     string $reloadCommand = "reloadMaterialXNode \"" + $nodeName + "\"";
     button -edit -command $editCommand editButton;
     button -edit -command $reloadCommand reloadButton;
 }
 
+global proc MaterialXSurface_FilePathNew(string $attrName)
+{
+    setUITemplate -pushTemplate attributeEditorTemplate;
+
+    rowLayout -numberOfColumns 3;
+        text -label "Document File Path";
+        textField documentFilePathField;
+        symbolButton -image "navButtonBrowse.xpm" documentFileBrowserButton;
+    setParent ..;
+    
+    MaterialXSurface_FilePathReplace($attrName);
+    setUITemplate -popTemplate;
+}
+
+global proc MaterialXSurface_FilePathReplace(string $attrName)
+{
+    evalDeferred ("connectControl documentFilePathField " + $attrName);
+    button -edit -command ("MaterialXSurface_FileBrowser " + $attrName) documentFileBrowserButton;
+}
+
+global proc MaterialXSurface_FileBrowser(string $attrName) 
+{
+    // Start in current file's directory if possible. 
+	string $sStartingdir = `textField -q -fileName documentFilePathField`;
```

Comment:
looks like you have a whitespace issue here.

Reply:
That's right, copied-and-pasted from another plugin. Fixed now.